google-cloud-apis/java-apis/maven-apis/src/com/google/cloud/tools/intellij/cloudapis/CloudApiUiExtension.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.module.Module;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Extension point for Cloud APIs, allowing access to UI and getting updates on cloud library
+ * selection, module selection, and final library selection confirmation. See also {@link
+ * CloudApiUiPresenter}.
+ */
+// TODO: move to core cloud API once dependency is inverted.
+public interface CloudApiUiExtension {
+  ExtensionPointName<CloudApiUiExtension> EP_NAME =
+      new ExtensionPointName<>("com.google.gct.cloudapis.cloudApiUiExtension");
+
+  /**
+   * Called when add cloud libraries dialog is created and opened and base UI is ready. At this
+   * point extension point is active.
+   *
+   * @param uiPresenter Presenter to access cloud API base UI.
+   */
+  void init(@NotNull CloudApiUiPresenter uiPresenter);
```

Comment:
what happens if the user fails to call init? will things be broken?

Reply:
init() will be called by core Cloud API module, and UI presenter will be passed in. This should not depend on sub-module. If you think this is not clear from javadoc or method name, I'll think a bit how to improve.

Reply:
this pattern made me thing of this: [http://go/tott-523](http://go/tott-523)

Reply:
Agree, not optimal. How about this:
1. Make `CloudApiUiPresenter` a service available from usual IDE service search.
2. Ensure all methods called on extension point have access to a valid UI presenter when callbacks are called.
2. Replace `init()` with `createCustomUiComponents()` (implemented in the next PR) - where UI components and necessary initialization is done by child sub-module?

Reply:
sgtm